pkg/frameworks/framework.go

```diff
@@ -0,0 +1,40 @@
+package frameworks
+
+import (
+	"fmt"
+
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/frameworks/custom"
+	"github.com/kubeflow/kfserving/pkg/frameworks/tensorflow"
+	v1 "k8s.io/api/core/v1"
+)
+
+type FrameworkHandler interface {
+	CreateModelServingContainer(modelName string) *v1.Container
+	ValidateSpec() error
+}
+
+func Get(modelSpec *v1alpha1.ModelSpec) (interface{ FrameworkHandler }, error) {
+	handlers := []interface{ FrameworkHandler }{}
```

Comment:
It seems like `Get` is doing both get and validation which is a bit confusing, maybe we can still leave the original validate method.

Reply:
So if Get doesn't do this check it will end up returning the first item that matches which is also incorrect... would it be okay if maybe it was named something other than get? Cant think of what offhand though..

Reply:
How about I call it MakeHandler (since it is actually making and not getting an existing thing)?